### ISSUE TITLE
Add More Examples

### ISSUE THREAD

**Issue by AtropineTears** [microsoft/Spartan#41]
I really would like if you added more examples of what you can do with Spartan. For instance, proving knowledge of a preimage of a hash with Poseidon.

This would be really helpful!

**Comment by srinathsetty** [comment]
Thanks, @AtropineTears for your interest! If you're familiar with bellman/bellperson libraries, adding a similar gadget library for Spartan is on our radar. For example, see https://github.com/microsoft/Nova/tree/main/src/bellperson that produces the necessary R1CS matrices from bellperson compatible gadgets. Would adding such a thing be helpful?

**Comment by AtropineTears** [comment]
Yes, if you guys can add more examples that would be helpful. I am still learning about ZKPs as they are quite a difficult subject to learn but they are quite interesting.

Spartan is a really great project that is ahead in its time in my personal opinion. The cubic example is good but it would be awesome if you guys could work on more gadgets. In the meantime,. I'll be looking at seeing if I can get some examples working.

Thank you

**Comment by srinathsetty** [comment]
We now have Spartan2 (https://github.com/Microsoft/Spartan2), with examples.
